src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -287,7 +288,8 @@ class SourcesTree extends Component<Props, State> {
         label: setDirectoryRootLabel,
         accesskey: setDirectoryRootKey,
         disabled: false,
-        click: () => setProjectDirectoryRoot(item.path)
+        click: () =>
+          this.props.setProjectDirectoryRoot(this.props.projectRoot, item.path)
```

Comment:
maybe we spread this.props before putting it into the clickhandler ... seems a bit hard to read as it is now.